flatdata-cpp/include/flatdata/internal/Reader.h

```diff
@@ -30,17 +30,31 @@ template < typename T, int offset = 0, int num_bits = sizeof( T ) * 8, int struc
 struct Reader
 {
     using StreamType = const unsigned char*;
-    using UnsignedType = typename BitsToUnsigned<
-        int_choice< num_bits, num_bits + offset % 8, num_bits + offset % 8 <= 64 >::value >::type;
+
     enum
     {
         bit_width = num_bits
     };
+    enum : typename UnderlyingType< T >::type
+    {
+        max = bit_width == 0 ? 0
+                             : ( bit_width - std::is_signed< T >::value
```

Comment:
Not the biggest fan of nesting ternary operators

Reply:
Any suggestions for how to improve this?

Reply:
Is it possible to do as an immediately executed lambda?
```
max = []()
{
    if ( condition )
    {... return value;}
    else
    {... return value;}
}();
```

Reply:
Not in C++11 (this needs to be constexpr)

Reply:
Hm, I can't think of anything else immediately